### PR TITLE
Attempt to fix Jacoco server running past test life cycle

### PR DESCRIPTION
**Pull request by abrooksv**
Attempt to make the jacoco remote server build service be re-entrant in case the daemon hasnt stopped it yet

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildSrc/src/main/kotlin/software/aws/toolkits/gradle/jacoco/RemoteCoverage.kt [hunk 1]
```diff
@@ -51,37 +61,41 @@ class RemoteCoverage private constructor(task: Test) {
...
-        private val fileWriter = ExecutionDataWriter(outputStream)
-        private val serverThread = Thread {
-            while (!signalShutdown.get()) {
-                val clientSocket = serverSocket.accept()
-                JacocoHandler(clientSocket, fileWriter).start()
+        private val isRunning = AtomicBoolean(false)
+
+        private val serverRunnable = Runnable {
+            parameters.execFile.asFile.get().outputStream().use {
+                while (!isRunning.get()) {
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply buildSrc/src/main/kotlin/software/aws/toolkits/gradle/jacoco/RemoteCoverage.kt:68]: Its initial state is false, but ya we should set it to false in the close

**Review by hunterwerlla — APPROVED** [review]
Looks like coverage works, hopefully it'll fix the root problem

**Pull request merged**